File: snippets/rust/src/fiat_currencies.rs
use std::sync::Arc;

use anyhow::Result;
use breez_liquid_sdk::prelude::*;

async fn list_supported_fiat_currencies(sdk: Arc<LiquidSdk>) -> Result<()> {
    // ANCHOR: list-fiat-currencies
    let supported_fiat_currencies = sdk.list_fiat_currencies().await?;
    // ANCHOR_END: list-fiat-currencies

    Ok(())
}

async fn get_current_rates(sdk: Arc<LiquidSdk>) -> Result<()> {
    // ANCHOR: fetch-fiat-rates
    let fiat_rates = sdk.fetch_fiat_rates().await?;
    // ANCHOR_END: fetch-fiat-rates

    Ok(())
}

File: snippets/go/fiat_currencies.go
package example

import (
	"log"

	"github.com/breez/breez-sdk-liquid-go/breez_sdk_liquid"
)

func ListFiatCurrencies(sdk *breez_sdk_liquid.BindingLiquidSdk) {
	// ANCHOR: list-fiat-currencies
	if fiatCurrencies, err := sdk.ListFiatCurrencies(); err == nil {
		log.Printf("%#v", fiatCurrencies)
	}
	// ANCHOR_END: list-fiat-currencies
}

func FetchFiatRates(sdk *breez_sdk_liquid.BindingLiquidSdk) {
	// ANCHOR: fetch-fiat-rates
	if fiatRates, err := sdk.FetchFiatRates(); err == nil {
		log.Printf("%#v", fiatRates)
	}
	// ANCHOR_END: fetch-fiat-rates
}
